### ISSUE TITLE
Parse Error when attempting update

### ISSUE THREAD

**Issue by AGenius** [lazydroid/auto-update-apk-client#71]
I have looked and looked for this issue and I am totally lost

Recently I get updates fail to install

When the update check happens I am notified there is an update . However when you select to get the update I am getting the following on all devices that try

Parse Error
There was a problem while parsing the package..

The logcat shows an entry

E/AutoUpdateApk: /data/data/com.[snippedout].oak/files/http:/www.autoupdateapk.com/fetch/AMIfv94BJqZmpJdzm_igbkkI2bD1hVOZwV7Nv2vtzUH1kL2_Kjwd827_JsJA6UN6YrP6uXi1dgWkL1thO9JcNUFaqBHaB8mLPWf-XldfNczabGWID-1QsDUm/Tr5KnZ5YEGIK9Ur2mbC0MZOpOvJWGlvmjXAostzvlsW-DR3LrBfpsUG6I2gIqi4/oak.apk: open failed: ENOENT (No such file or directory)

It used to work but no longer since a month now

What could be wrong

App is compiled to v5 api

**Comment by mlanzone** [comment]
I have the same issue.....have tried for DAYS to contact site owner but no reply.....if I do not hear back soon I will have to unsubscribe....such a shame :(

**Comment by eboateng1** [comment]
I also have the same problem. Have been attempting to contact them as well. Nothing at all. 
Really unreliable support. Would understand if this project was discontinued but their is no notification or details leading to it. As well as they are still getting paid for this......

**Comment by leenolasco** [comment]
I thought i was the only one experiencing this kind of problem, looks like they wont continue to support this.

**Comment by mlanzone** [comment]
guess I should "pull down" my apps and cancel my subscription :(

**Comment by shrutijoshigithub1** [comment]
Facing the same issue..

**Comment by AGenius** [comment]
I was up upto 3am last night/morning digging into this.

It appears it might be due to out of date functionality

The notification usage is depreciated, it gets the details that there is a update, and a download link (very long url), if I paste this into a browser it downloads, however it fails to start the download and I suspect this is because its using out dated HTTP functions...

I tried to fix it but was struggling. I even redirected the download to pull from my own web server instead of from autoupdate server and it still failed. It fails because it cannot start a download using the HTTP methods ...  the client is no good anymore for newer api.

I am pulling my app and looking to use the google play private options maybe

I am still developing my first application and its not completely ready for full deploy to the client so I have not setup any subscription yet , not going to bother now......

**Comment by mlanzone** [comment]
I gave up also...
Ended up writing my own Auto Updater .

It is working GREAT!

**Comment by shrutijoshigithub1** [comment]
Could you please share your thoughts/codes of your own Auto Updater?

**Comment by mlanzone** [comment]
I have my own servers.
So I now host my own "Auto-Update" system.

**Comment by shrutijoshigithub1** [comment]
ok..

**Comment by AGenius** [comment]
@mlanzone care to share ? going to pull this one and look for alternatives

**Comment by mlanzone** [comment]
what would u like to know ?

**Comment by AGenius** [comment]
everything

How do you check for your update, how do you pull / notify like this one did?

**Comment by mlanzone** [comment]
pls reply to mikel@supportresourcesinc.com
:)

**Comment by AGenius** [comment]
Found the reason at last... the get of the package from the autoupdateapk.com site returns a Internal server error when the file is attempted to be downloaded

I modified to pull from my own server and ignore the content type check and it worked.  But along with this I changed the code in SetupVars to ALWAYS remove the settings of the downloaded when the check happens because the notification was always trying to launch a non existent file due to the previous failures.

So in summary there is a problem on the server. no idea what it is because no one there has bothered to respond to the support request to be able to look at the log files and see what it might be... something wrong with the manifest in the apk . no idea what it might be

suffice to say , I will NOT be using this service anymore .

**Comment by AGenius** [comment]
I have now got my own server for version checking and updating and it works perfectly

Bye AutoUpdateAPK!!!!!!!!!!!!!!!!!!

**Comment by maoxiong** [comment]
Ok for anyone still using this - it seems like it might be a permissions problem.

If you try to execute the file in your apps /files folder, it won't run (even externally) - but move the same file into a public area (say your download folder) and it will now execute.

**Comment by maoxiong** [comment]
Ok, to solve it:

Search for all instances in the files of:
`context.getFilesDir().getAbsolutePath()`

And change to:
`Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS)`

This will now use the downloads folder for the download (file gets deleted after)

Also need to edit the checkUpdateTask function.

Change:
`FileOutputStream out = context.openFileOutput( fname, Context.MODE_WORLD_READABLE);`

to:
`FileOutputStream out = new FileOutputStream(Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS) + "/" + fname);`

And you're done. Should work fine now :)

**Comment by mlanzone** [comment]
how do I get off this thread.

I have already resolved the issue by creating my own Auto-Update.
